Tasks/src/provider/Provider.java

```diff
@@ -3,11 +3,24 @@
 import java.util.ArrayList;
 
 class Provider {
-    public void provide() {
+    public void provideList() {
         Consumer.consume(new ArrayList<B>());
     }
 
-    public void provideB() {
+    public void provide() {
         Consumer.consume(new B());
     }
+
+    public void provideArray() {
+        final B[] b = new B[]{ new B(), new B()};
+
+        //error here
+        b[1] = new C();
+
+        //but there is no error at compile time, but error at runtime!
+        final A[] a = b;
+        a[1] = new C();
+
```

Comment:
Here in example we added instance of `C` into `B` array. So, because of arrays in Java are covariant, we can get round type control. And `List<T>` in Java is invariant.